pattern_library/loader_tags.py

```diff
@@ -150,3 +150,90 @@ def do_include(parser, token):
         extra_context=namemap,
         isolated_context=isolated_context,
     )
+
+def visit_extends(self, node, frame):
+    """This method serves as overriding the jinja extends tag
+    Dupe of the jinja2.compiler.CodeGenerator visit_Extends
+    except for
+        self.writeline(
+            "parent_template.new_context(context.get_all(), True,"
+            f" {self.dump_local_context(frame)})"
+        )
+    which executes at runtime to pull in the dpl context
+    Handles visiting extends
+    """
+    from jinja2.compiler import CompilerExit
+
+    if not frame.toplevel:
+        self.fail("cannot use extend from a non top-level scope", node.lineno)
+    # if the number of extends statements in general is zero so
+    # far, we don't have to add a check if something extended
+    # the template before this one.
+    if self.extends_so_far > 0:
+        # if we have a known extends we just add a template runtime
+        # error into the generated code.  We could catch that at compile
+        # time too, but i welcome it not to confuse users by throwing the
+        # same error at different times just "because we can".
+        if not self.has_known_extends:
+            self.writeline("if parent_template is not None:")
+            self.indent()
+        self.writeline('raise TemplateRuntimeError("extended multiple times")')
+
+        # if we have a known extends already we don't need that code here
+        # as we know that the template execution will end here.
+        if self.has_known_extends:
+            raise CompilerExit()
+        else:
+            self.outdent()
+    self.writeline("parent_template = environment.get_template(", node)
+    self.visit(node.template, frame)
+    self.write(f", {self.name!r})")
+    # addition to update the context with dpl context
+    # calls the template_new_context method below when
+    # invoked at runtime
+    self.writeline(
+        "parent_template.new_context(context.get_all(), True,"
+        f" {self.dump_local_context(frame)})"
+    )
+    self.writeline("for name, parent_block in parent_template.blocks.items():")
```

Comment:
does this _have_ to be here vs at the end? ie could we do the default visit_  and then outdent and setdefault on the context that's already defined? replicating the whole function here vs wrapping it feels very fragile